### PR TITLE
Overhaul

### PR DESCRIPTION
**Pull request by camsom**
@MichaelButkovic @mparent61 

Rewrote everything to be a bit simpler

### CODE REVIEW

**Comment by MichaelButkovic** [comment]
Can this be added to Travis?

**Comment by camsom** [comment]
There you go. Let me die

**Comment by MichaelButkovic** [comment]
:smile:

File: parent_swap/swap.py [hunk 1]
```diff
@@ -1,22 +1,43 @@
...
+    default_model_name = getattr(settings, 'DEFAULT_BASE_MODEL', None)
+    return import_class(default_model_name, models.Model)
+
+
+def get_base_mapping(base_cls):
+    return getattr(base_cls, 'Mapping', object)
+
+
+def get_base_serializer():
+    default_serializer_name = getattr(settings, 'DEFAULT_SERIALIZER', None)
```
Comment (mparent61) [inline parent_swap/swap.py:31]: Seems like all of these settings should all have _BASE_ in their name to be consistent.

**Comment by mparent61** [comment]
:+1:

**Pull request merged**